background.ts

```diff
@@ -1052,18 +1059,32 @@ class BackgroundCommandWorker implements CommandWorkerInterface {
     })());
   }
 
+  async getExtensionManager() {
+    const getEM = (await import('@pkg/main/extensions/manager')).default;
+
+    return await getEM();
+  }
+
   listExtensions(): Promise<Record<string, true>> {
     const entries = Object.entries(cfg.extensions).filter(([k, v]) => v) as [string, true][];
 
     return Promise.resolve(Object.fromEntries(entries));
   }
 
+  async listExtensionsMetadata() {
+    const extensionManager = await this.getExtensionManager();
+    const extensions = await extensionManager?.getExtensions();
+    const installedExtensions = Object.keys(await this.listExtensions());
+    const filteredExtensions = extensions?.filter(x => installedExtensions.includes(x.id));
+
+    window.send('extensions-list', filteredExtensions);
+  }
+
   async installExtension(id: string, state: 'install' | 'uninstall'): Promise<{status: number, data?: any}> {
-    const getEM = (await import('@pkg/main/extensions/manager')).default;
-    const em = await getEM();
+    const em = await this.getExtensionManager();
     const extension = em?.getExtension(id);
 
-    if (!extension) {
+    if (!extension || typeof (em) === 'undefined') {
```

Comment:
If `em` is undefined, then `em?.<stuff>` is undefined, so `extension` should be falsy, right? Why is the extra check needed? (I may be missing something, in which case a comment would be useful.)